cmake/CompilerConfiguration.cmake

```diff
@@ -41,8 +41,8 @@ Please open an issue at https://github.com/cp2k/dbcsr/issues with the reported c
 endif ()
 
 if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-  set(CMAKE_CXX_FLAGS_RELEASE      "-O3 -g -funroll-loops -Wall -Werror")
-  set(CMAKE_CXX_FLAGS_COVERAGE     "-O0 -g --coverage -Wall -Werror")
+  set(CMAKE_CXX_FLAGS_RELEASE      "-O3 -g -funroll-loops -Wall -Werror -Wno-error=deprecated-declarations")
+  set(CMAKE_CXX_FLAGS_COVERAGE     "-O0 -g --coverage -Wall -Werror -Wno-error=deprecated-declarations")
```

Comment:
We should drop the `-Werror` for the release.
And I don't like having `-Wno-error` on the whole project for a subcomponent.

Reply:
>  And I don't like having -Wno-error on the whole project for a subcomponent.
`set_source_files_properties` for one specific source file then?

Reply:
```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
...
#pragma GCC diagnostic pop
```
annotations is another way to do it...

Reply:
sounds good, with a generator expression `$<CXX_COMPILER_ID:GNU>`?

Reply:
... also the `#pragma` works, might be even cleaner.

Reply:
pragma it is